src/axom/sidre/core/MFEMSidreDataCollection.cpp

```diff
@@ -1206,13 +1277,100 @@ void MFEMSidreDataCollection::RegisterField(const std::string& field_name,
   // Register field_name in the blueprint_index group.
   if(myid == 0)
   {
-    RegisterFieldInBPIndex(field_name, gf);
+    RegisterFieldInBPIndex(field_name, gf->VectorDim());
   }
 
   // Register field_name + gf in field_map.
   DataCollection::RegisterField(field_name, gf);
 }
 
+void MFEMSidreDataCollection::RegisterQField(const std::string& field_name,
+                                             mfem::QuadratureFunction* qf,
+                                             const std::string& buffer_name,
+                                             axom::sidre::IndexType offset)
+{
+  if(field_name.empty() || buffer_name.empty() || qf == NULL ||
+     qf->GetSpace() == NULL)
+  {
+  #ifdef AXOM_DEBUG
+    SLIC_WARNING("QField with the name '"
+                 << field_name
+                 << "' was provided a null"
+                    " QuadratureFunction, so nothing was done.");
+  #endif
+    return;
+  }
+
+  // Register field_name in the blueprint group.
+  sidre::Group* f = m_bp_grp->getGroup("fields");
+
+  if(f->hasGroup(field_name))
+  {
+    // There are two possibilities:
+    // 1. If HasQField(field_name) is true - we are overwriting a field that
+    //    was previously registered.
+    // 2. Otherwise, the field was loaded from a file, or defined outside of
+    //    the data collection.
+    if(HasQField(field_name))
+    {
+  #ifdef AXOM_DEBUG
+      // Warn about overwriting field.
+      // Skip warning when re-registering the nodal grid function
+      if(field_name != m_meshNodesGFName)
+      {
+        SLIC_WARNING("QField with the name '"
+                     << field_name
+                     << "' is already "
+                        "registered, overwriting the old qfield");
+      }
+  #endif
+      DeregisterQField(field_name);
+    }
+  }
+
+  // This will return the existing field (if external), otherwise, a new group
+  sidre::Group* grp = alloc_group(f, field_name);
+
+  // A QField has the following schema:
+  // <m_bp_grp>/fields/<field_name>/
+  // <m_bp_grp>/fields/<field_name>/topology     = "mesh"
+  // <m_bp_grp>/fields/<field_name>/values       = array of size QuadratureFunction::Size()
+  // <m_bp_grp>/fields/<field_name>/basis        = string that encodes the QF
+  //                                               integration order and vector dimension
+  //                                               "QF_Default_[ORDER]_[VDIM]"
+
+  // Set the "basis" string using the qf's order and vdim, overwrite if
+  // necessary.
+  sidre::View* v = alloc_view(grp, "basis");
+  // In the future, we should be able to have mfem::QuadratureFunction provide us this name.
+  const std::string basis_name =
+    fmt::format("QF_Default_{0}_{1}", qf->GetSpace()->GetOrder(), qf->GetVDim());
```

Comment:
```suggestion
    fmt::format("QF_Default_{0}_{1}", qf->GetSpace()->GetElementIntRule(0).GetOrder(), qf->GetVDim());
```

This should be all that's needed to run with MFEM v4.2 and get around the newer getter function that was added in between MFEM 4.2 and 4.3.

I've tried compiling this locally on my copy of axom. However, I've been running into some issues with it not picking up my copy of hdf5 :/ after my system updated, which has caused the build to fail earlier on...